smartsim/_core/utils/helpers.py

```diff
@@ -38,28 +38,28 @@
 
 
 def unpack_db_identifier(db_id: str, token: str) -> t.Tuple[str, str]:
-    """Unpack the unformatted database identifier using the token,
-    and format for env variable suffix
-    :db_id: the unformatted database identifier eg. identifier_1_0
-    :token: character '_' or '-' to use to unpack the database identifier
-    :return: db suffix, and formatted db_id eg. _identifier_1, identifier_1
+    """Unpack the unformatted database identifier
+    and format for env variable suffix using the token
+    :param db_id: the unformatted database identifier eg. identifier_1
+    :type db_id: str
+    :param token: character to use to construct the db suffix
+    :type token: str
+    :return: db id suffix and formatted db_id e.g. ("_identifier_1", "identifier_1")
+    :rtype: (str, str)
     """
 
     if db_id == "orchestrator":
         return "", ""
-    db_id = "_".join(db_id.split(token)[:-1])
-    # if unpacked db_id is default, return empty
-    if db_id == "orchestrator":
-        # if db_id is default after unpack, return empty
-        return "", ""
-    db_name_suffix = "_" + db_id
+    db_name_suffix = token + db_id
     return db_name_suffix, db_id
```

Comment:
Yeah, so, the original method was taking in the `Orchestrator` name, which was assumed to be along the lines of `db_identifier + "_<id>"` but that was prone to wrong parsing, for example when the user gave a name like `my_first_db` and we did not catch it.
The whole goal of this method is to return either the orchestrator id or `<token>db_identifier` which has to be appended to `SSDB` var. If the name is `"orchestrator"` then, for backward compatibility, it returns `""` for both, because we just need `SSDB` without suffix.
Admittedly, it has become such a small function that we can easily add it to the `Orchestrator` class as a member.

Reply:
Ahhh I see! I wonder if this is actually evidence that the `db_identifier` property _shouldn't_ actually be the same as the name. Something more to the nature of:
```py
orc = Orchestrator()
assert orc.name == "orchestrator"
assert orc.db_identifier == ""

orc2 = Orchestrator(db_identifier="my_first_db")
assert orc2.name == "orchestrator_my_first_db"
assert orc2.db_identifier == "my_first_db"
```
where the _name_ is actually the computed property, to the nature of
```py
class Orchestrator(...):
    ...

    @property
    def name(self) -> str:
        return f"orchestrator_{self.db_identifier}"

    ...
```
But this is definitely out of scope for this PR! Lets circle back on this when we start looking into a large scale Orchestrator refactor!

Reply:
Yah, so the other aspect of this that Julia was struggling with was that there's a few places where we have different conventions for how we refer to the orchestrator name. It's definitely something that needs to be addressed in the future.